src/Dapr.AspNetCore/DaprEndpointRouteBuilderExtensions.cs

```diff
@@ -78,5 +77,67 @@ public static IEndpointConventionBuilder MapSubscribeHandler(this IEndpointRoute
                 await writer.FlushAsync();
             });
         }
+
+        /// <summary>
+        /// Maps an endpoint that will respond to requests to <c>/dapr/subscribe</c> from the
+        /// Dapr runtime.
+        /// </summary>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder" />.</param>
+        /// <param name="options">Configuration options</param>
+        /// <returns>The <see cref="IEndpointConventionBuilder" />.</returns>
+        /// <seealso cref="MapSubscribeHandler(IEndpointRouteBuilder)"/>
+        public static IEndpointConventionBuilder MapSubscribeHandler(this IEndpointRouteBuilder endpoints, SubscribeOptions options)
+        {
+            if (endpoints is null)
+            {
+                throw new System.ArgumentNullException(nameof(endpoints));
+            }
+
+            return endpoints.MapGet("dapr/subscribe", async context =>
+            {
+                var dataSource = context.RequestServices.GetRequiredService<EndpointDataSource>();
+                var entries = dataSource.Endpoints
+                    .OfType<RouteEndpoint>()
+                    .Where(e => e.Metadata.GetMetadata<ITopicMetadata>()?.Name != null)   // only endpoints which have  TopicAttribute with not null Name.
+                    .Distinct()
+                    .Select(e => (e.Metadata.GetMetadata<ITopicMetadata>().PubsubName, e.Metadata.GetMetadata<ITopicMetadata>().Name, e.Metadata.GetMetadata<IRawTopicMetadata>()?.EnableRawPayload, e.RoutePattern));
+
+
+                var logger = context.RequestServices.GetService<ILoggerFactory>().CreateLogger("DaprTopicSubscription");
+
+                ArrayList subscriptions = new();
```

Comment:
We should not use `ArrayList`. I'd suggest either using a strongly-typed collection like `List<>` or doing the approach used in the other method (just loop directly with no intermediate list).